### PR TITLE
Feature/book tests

### PR DESCRIPTION
**Pull request by ReamFOX**
codeCoverage

### CODE REVIEW

File: src/test/java/com/farion/onlinebookstore/controller/BookControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        int testId = 5;
+
+        MvcResult result = mockMvc.perform(
+                        MockMvcRequestBuilders.get(BOOK_ENDPOINT + SLASH + testId)
+                )
+                .andExpect(status().is2xxSuccessful()).andReturn();
+        BookDto actual =
+                objectMapper.readValue(result.getResponse().getContentAsString(), BookDto.class);
+
+        EqualsBuilder.reflectionEquals(booksFromDB.get(testId), actual);
```
Comment (andrii-hoienko) [inline src/test/java/com/farion/onlinebookstore/controller/BookControllerTest.java]:
```suggestion
        int testId = 5;
        BookDto expected= ...
        

        MvcResult result = mockMvc.perform(MockMvcRequestBuilders.get(BOOK_ENDPOINT + SLASH + testId))
                .andExpect(status().is2xxSuccessful())
                .andReturn();
                
        BookDto actual =
                objectMapper.readValue(result.getResponse().getContentAsString(), BookDto.class);

        EqualsBuilder.reflectionEquals(booksFromDB.get(testId), actual);
```

File: src/test/java/com/farion/onlinebookstore/controller/BookControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    void findBookByTitle_validData_EmptyResponse() throws Exception {
+        String testTitle = "Microeconomic";
+
+        MvcResult result = mockMvc.perform(MockMvcRequestBuilders.get(
+                BOOK_ENDPOINT + SLASH + SEARCH_BY_TITLE + testTitle))
+                .andExpect(status().isOk())
+                .andReturn();
+        List<BookDto> actual = objectMapper.readValue(result.getResponse().getContentAsString(),
+                new TypeReference<>() {
+                });
```
Comment (andrii-hoienko) [inline src/test/java/com/farion/onlinebookstore/controller/BookControllerTest.java]: What is the reason to write TypeReference?

File: src/test/java/com/farion/onlinebookstore/controller/BookControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+        MvcResult result = mockMvc.perform(MockMvcRequestBuilders.get(BOOK_ENDPOINT))
+                .andExpect(status().is2xxSuccessful()).andReturn();
+        List<BookDto> actual = objectMapper.readValue(result.getResponse().getContentAsString(),
+                new TypeReference<>() {
+                });
+
+        for (int i = 0; i < actual.size(); i++) {
+            EqualsBuilder.reflectionEquals(expected.get(i), actual.get(i));
+        }
```
Comment (andrii-hoienko) [inline src/test/java/com/farion/onlinebookstore/controller/BookControllerTest.java]: I suggest to move this logic to util class and reuse when required, or use library that allows to assert collections

File: src/test/java/com/farion/onlinebookstore/controller/CategoryControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+                        + "but the feeling they inspire in the reader.";
+        CreateCategoryRequestDto requestDto = new CreateCategoryRequestDto();
+        requestDto.setName(expectedName);
+        requestDto.setDescription(expectedDescription);
+
+        mockMvc.perform(post(CATEGORY_ENDPOINT).contentType(MediaType.APPLICATION_JSON)
+                        .content(objectMapper.writeValueAsString(requestDto)))
+                .andExpect(status().isCreated()).andExpect(jsonPath("$.name").value(expectedName))
+                .andExpect(jsonPath("$.description").value(expectedDescription));
+    }
```
Comment (andrii-hoienko) [inline src/test/java/com/farion/onlinebookstore/controller/CategoryControllerTest.java]: I suggest to parse object received from  mockMvc.perform to business object and compare using asserEquals

File: src/test/java/com/farion/onlinebookstore/util/ConstUtil.java [hunk 5]
```diff
@@ -0,0 +1,27 @@
+package com.farion.onlinebookstore.util;
+
+import com.farion.onlinebookstore.entity.Category;
+import java.io.File;
+import java.util.List;
+import java.util.Set;
+
+public class ConstUtil {
```
Comment (andrii-hoienko) [inline src/test/java/com/farion/onlinebookstore/util/ConstUtil.java]:
```suggestion
public class Constants { //TestObjectMother as an alternative name, and you can move here methods that create test objects 
```

File: src/test/java/com/farion/onlinebookstore/util/ConstUtil.java [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    public static final File TEST_BOOKS =
+            new File("src/test/resources/data/test-books.json");
+    public static final File TEST_CATEGORIES =
+            new File("src/test/resources/data/test-categories.json");
+    public static final Long TEST_ID = 1L;
+    public static final Set<Long> TEST_CATEGORY = Set.of(1L);
+    public static final List<Category> TEST_CATEGORY_LIST = List.of(new Category(), new Category());
+
+    private ConstUtil() {
+        throw new RuntimeException("This constructor shouldn`t invoke");
```
Comment (andrii-hoienko) [inline src/test/java/com/farion/onlinebookstore/util/ConstUtil.java]:
```suggestion

```

**Review by andrii-hoienko — CHANGES_REQUESTED** [review]
Good job! Left some comments

File: src/test/java/com/farion/onlinebookstore/controller/BookControllerTest.java (hunk 2 above)
Comment (ReamFOX) [reply src/test/java/com/farion/onlinebookstore/controller/BookControllerTest.java]: I found this as an option to use a mapper for a collection, but I can also use another option.

**Review by vovasalyha — APPROVED** [review]

**Pull request merged**